### PR TITLE
Remove obsolete properties, methods, and classes for 3.0

### PR DESCRIPTION
**Pull request by erikmav**
Add BinaryFormattter serialize-deserialze testing across versions, and correct serialization property bag names broken by modernization (#124).

Commented out obsolete tag for FixFlags.Mdc as the obsolete tags in MDC.cs are commented out.

### CODE REVIEW

**Comment by erikmav** [comment]
For #125

**Review by FreeAndNil — COMMENTED** [review]
Thanks a lot.
Just one question.

I've noted the breaking changes in https://github.com/apache/logging-log4net/issues/125#issuecomment-2034645504

File: src/log4net/Config/RepositoryAttribute.cs [hunk 1]
```diff
@@ -43,12 +43,8 @@ namespace log4net.Config
   /// <author>Gert Driesen</author>
   [AttributeUsage(AttributeTargets.Assembly)]
   [Serializable]
-  public /*sealed*/ class RepositoryAttribute : Attribute
+  public sealed class RepositoryAttribute : Attribute
```
[the remark this answers is not in the document]
Comment (erikmav) [reply src/log4net/Config/RepositoryAttribute.cs:46]: Fixed. This was victim of various attempts to seal the complex hierarchy of attributes.

**Review by FreeAndNil — APPROVED** [review]
Thanks.

**Comment by FreeAndNil** [comment]
@erikmav because of https://learn.microsoft.com/en-us/dotnet/standard/serialization/binaryformatter-security-guide we think about removing ISerializable and [Serializable] (at least for netstandard).
What do you think?
Where are you using this serialization for log4net objects?

**Comment by erikmav** [comment]
Unneeded by me. During the 3.0 work it was not clear that serialization could be removed, so I added the newer extensive unit tests including ensuring back compat. Removing it along with all the tests is fine by me.
________________________________

**Comment by FreeAndNil** [comment]
@erikmav thanks for your feedback.
We have only three failing tests, when removing it.
The two from you and PropertiesDictionaryTest.
When we remove the old serialization we will maybe switch to DataContractSerializer and adjust the tests.

**Comment by vy** [comment]
Hello @erikmav! Log4net crew is actively looking for user feedback and, new members. :wink: @FreeAndNil and I would really appreciate it if we can have a video call with you at your convenience. If that sounds like something you want to participate, would you mind [reaching out to me with an email](mailto:volkan@yazi.ci), please?

**Pull request merged**